src/mediapipe_internal/mediapipegraphexecutor.cpp

```diff
@@ -165,55 +179,95 @@ Status MediapipeGraphExecutor::infer(const KFSRequest* request, KFSResponse* res
         SPDLOG_DEBUG("[servable name: {} version: {}] Invalid number of inputs - {}", request->model_name(), version, details);
         return Status(StatusCode::INVALID_NO_OF_INPUTS, details);
     }
-    for (auto name : inputNames) {
-        SPDLOG_DEBUG("Tensor to deserialize:\"{}\"", name);
-        ov::Tensor input_tensor;
-        auto status = deserializeTensor(name, version, request, input_tensor);
-        if (!status.ok()) {
-            SPDLOG_DEBUG("Failed to deserialize tensor: {}", name);
-            return status;
+
+    ::mediapipe::Packet packet;
+    std::set<std::string> outputPollersWithReceivedPacket;
+
+    // Passing whole KFS request and response
+    if (this->passKfsRequestFlag) {
+        SPDLOG_DEBUG("Passing whole KFS request and response");
+        for (auto name : inputNames) {
```

Comment:
Unless you want to refactor/templatize code here with nonKFSpassthrough then the loop is not needed here. We know we have just one input and output.